src/System.Net.WebSockets.Client/src/System/Net/WebSockets/ClientWebSocket.cs

```diff
@@ -146,6 +144,31 @@ public Task ConnectAsync(Uri uri, CancellationToken cancellationToken)
             return ConnectAsyncCore(uri, cancellationToken);
         }
 
+        private async Task ConnectAsyncCore(Uri uri, CancellationToken cancellationToken)
+        {
+            _innerWebSocket = WebSocketHandle.Create();
+
+            try
+            {
+                // Change internal state to 'connected' to enable the other methods
+                if ((InternalState)Interlocked.CompareExchange(ref _state, (int)InternalState.Connected, (int)InternalState.Connecting) != InternalState.Connecting)
+                {
+                    // Aborted/Disposed during connect.
+                    throw new ObjectDisposedException(GetType().FullName);
+                }
+
+                await _innerWebSocket.ConnectAsyncCore(uri, cancellationToken, _options);
+            }
+            catch (Exception ex)
+            {
+                if (Logging.On)
+                {
+                    Logging.Exception(Logging.WebSockets, this, "ConnectAsync", ex);
+                }
+                throw;
```

Comment:
Ah, after reading further, I see that you only moved this logic, so this question isn't specific to this PR.